ext/pdo_pgsql/pgsql_statement.c

```diff
@@ -705,6 +705,26 @@ static int pdo_pgsql_stmt_cursor_closer(pdo_stmt_t *stmt)
 	return 1;
 }
 
+static int pgsql_stmt_get_attr(pdo_stmt_t *stmt, zend_long attr, zval *val)
+{
+	pdo_pgsql_stmt *S = (pdo_pgsql_stmt*)stmt->driver_data;
+
+	switch (attr) {
+#ifdef HAVE_PG_RESULT_MEMORY_SIZE
+		case PRO_PGSQL_ATTR_RESULT_MEMORY_SIZE:
+			if(stmt->executed) {
+				ZVAL_LONG(val, PQresultMemorySize(S->result));
+			} else {
+				ZVAL_NULL(val);
```

Comment:
I do not know if we should emit a PDO exception in this case, @SakiTakamachi what do you think ?

Reply:
I think we can throw an exception here, because it's obviously wrong to try to get the size of a result when there is no "result" yet.

Reply:
@devnexen 

Ah but mainly these are used for debugging purposes?

In that case, throwing an exception might make the method difficult to use.

If it's unexecuted, does it make sense to interpret this as 0 memory allocation and return 0?

Reply:
IMO, I prefer to throw an exception for incorrect operations.

On the other hand, if you `fetchAll()` a statement that has not yet been executed, `[]` is returned. If consistency is our priority, it may be better not to throw exceptions.

In both cases, it doesn't take me much time since I have the modified code at hand as well.

Reply:
Rather, it seems like the problem is that fetching an unexecuted result doesn't result in an error....

That's a relic of PDO when exceptions didn't exist, and I don't think it should still be that way if don't take BC into account.

I'll wait for David's opinion.

Reply:
I still think about it and wrestle with the idea ; we can take our time no rush to merge this functionality :)

Reply:
What is also possible is to fill out pdo_odbc_errinfo `last_state` and `last_err_msg` fields to at least warn the caller.

Reply:
One of the least invasive approach might be to use `pdo_pgsql_error_stmt_msg` with the 3rd argument as NULL or as `HY000`.

Reply:
How about `pdo_raise_impl_error`? SQL status seems to be a bit of a noise here.

Reply:
Ah, excuse me. This also has a status....

Reply:
I added `pdo_pgsql_error_stmt_msg` in https://github.com/php/php-src/pull/14260/commits/da6b54b5d297e0f0fd7f603ab37d882881ca12b0.

If we should not emit exception, I think the current code is the best.

Reply:
You can even be more informative since you have access to the statement's name.
Note that, with the last change, your new test is skipped due to the new constant now missing.

Reply:
oh I see, it works only if you enable pgsql too.

Reply:
Thank you for your time! ! !
I fixed:
* [add a statement name to errorInfo()](https://github.com/php/php-src/pull/14260/commits/c58707d576739cd741d105b8cbc9c559713a3430)
* [constant declaration location](https://github.com/php/php-src/pull/14260/commits/37e203e91d35dbcfedc20c67b40933c2482f6f71)